pandas/_libs/tslibs/parsing.pyx

```diff
@@ -1016,9 +1016,11 @@ def guess_datetime_format(dt_str: str, bint dayfirst=False) -> str | None:
 
 cdef str _fill_token(token: str, padding: int):
     cdef str token_filled
-    if "." not in token:
+    if re.search(r"\d+\.\d+", token) is None:
+        # For example: 98
         token_filled = token.zfill(padding)
     else:
+        # For example: 00.123
         seconds, nanoseconds = token.split(".")
```

Comment:
is there a risk that this raises if there is a "." but not the regex?

Reply:
yup, you'd get
```python
In [6]: token = '.'

In [7]: seconds, nanoseconds = token.split(".")

In [8]: seconds = f"{int(seconds):02d}"
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In [8], line 1
----> 1 seconds = f"{int(seconds):02d}"

ValueError: invalid literal for int() with base 10: ''
```

which is what happens in the string in the linked issue (`'27.03.2003 14:55:00.000'`)

Reply:
yeah, if the padding was `2` (say) then `'.'.zfill(2)` would give `'0.'`, and in

https://github.com/pandas-dev/pandas/blob/f7979be99585030a17f4d4a41f7447f11a9e1660/pandas/_libs/tslibs/parsing.pyx#L945-L946

it wouldn't match `parsed_formatted`